cmd/influxd/run.go

```diff
@@ -159,25 +159,18 @@ func Run(config *Config, join, version string, logWriter *os.File) (*messaging.B
 			parser.Separator = c.NameSeparatorString()
 			parser.LastEnabled = c.LastEnabled()
 
-			// Start the relevant server.
-			if strings.ToLower(c.Protocol) == "tcp" {
-				g := graphite.NewTCPServer(parser, s)
-				g.Database = c.Database
-				g.SetLogOutput(logWriter)
-				err := g.ListenAndServe(c.ConnectionString(config.BindAddress))
-				if err != nil {
-					log.Printf("failed to start TCP Graphite Server: %v\n", err.Error())
-				}
-			} else if strings.ToLower(c.Protocol) == "udp" {
-				g := graphite.NewUDPServer(parser, s)
-				g.Database = c.Database
-				g.SetLogOutput(logWriter)
-				err := g.ListenAndServe(c.ConnectionString(config.BindAddress))
-				if err != nil {
-					log.Printf("failed to start UDP Graphite Server: %v\n", err.Error())
-				}
-			} else {
-				log.Fatalf("unrecognized Graphite Server protocol %s", c.Protocol)
+			// Spin up the server.
+			var g graphite.Server
+			g, err := graphite.NewServer(c.Protocol, parser, s)
+			if err != nil {
+				log.Fatalf("failed to initialize %s Graphite server: %s", c.Protocol, err.Error())
+			}
+
+			g.SetDatabase(c.Database)
+			g.SetLogOutput(logWriter)
```

Comment:
Could these two be factored out and put in as arguments to the NewServer function?